did/model.go

```diff
@@ -12,6 +12,32 @@ const (
 	KnownDIDContext string = "https://www.w3.org/ns/did/v1"
 )
 
+// https://www.w3.org/TR/did-core/#did-document-metadata
+type DIDDocumentMetadata struct {
+	Created       string `json:"created,omitempty"`
+	Updated       string `json:"updated,omitempty"`
+	Deactivated   bool   `json:"deactivated"`
```

Comment:
consider adding a validation tag for this, which will enable us to use the go validation lib

```suggestion
	Created       string `json:"created,omitempty" validate:"datetime"`
   Updated string `json:"updated,omitempty" validate:"datetime"`
	Deactivated   bool   `json:"deactivated" validate:"required"`
```

there are a number of possible validations, I haven't tried them all...but might be worth looking into

and follow up with an `IsValid()` method for the struct

https://github.com/go-playground/validator

Reply:
good callout.